modules/apps/content-dashboard/content-dashboard-web/test/js/components/Sidebar.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {act, cleanup, render} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React, {useState} from 'react';
+
+import Sidebar from '../../../src/main/resources/META-INF/resources/js/components/Sidebar';
+
+import '@testing-library/jest-dom/extend-expect';
+
+const ControlledSidebar = ({open, subtitle = '', title = ''}) => {
+	const [isOpen, setIsOpen] = useState(open);
+
+	return (
+		<Sidebar onClose={() => setIsOpen(false)} open={isOpen}>
+			<Sidebar.Header subtitle={subtitle} title={title} />
+		</Sidebar>
+	);
+};
+
+describe('Sidebar', () => {
+	beforeEach(() => {
+		jest.useFakeTimers();
+	});
+
+	afterEach(() => {
+		jest.clearAllTimers();
+		cleanup();
+	});
+
+	it('renders an open sidebar', () => {
+		const {asFragment} = render(<Sidebar />);
+
+		act(() => {
+			jest.runAllTimers();
+		});
+
+		expect(asFragment()).toMatchSnapshot();
```

Comment:
Out of curiosity, how come you chose to use `asFragment` instead of just `container`? This is the first time I've seen `asFragment`, so I'm not totally aware of the tradeoffs

Reply:
TBH I'm not familiar with testing-library/react, I just copied the patter from other tests. I'll take a look to find if this is for a reason or if I can use container. Thx!!

Reply:
```js
expect(asFragment()).toMatchSnapshot();
```
Wraps the HTML inside a `<DocumentFragment>`

```js
expect(container).toMatchSnapshot();
```
Wraps the HTML inside a `<div>`

I saw both in the portal. Otherwise, if I'm honest, I don't know what's the reason... 🤔